server/routes/surveyRoutes.js

```diff
@@ -23,35 +24,8 @@ module.exports = app => {
   });
 
   app.post('/api/surveys/webhooks', (req, res) => {
-    const p = new Path('/api/surveys/:surveyId/:choice');
-
-    _.chain(req.body)
-      .map(({ email, url }) => {
-        const match = p.test(new URL(url).pathname);
-        if (match) {
-          return { email, surveyId: match.surveyId, choice: match.choice };
-        }
-      })
-      .compact()
-      .uniqBy('email', 'surveyId')
-      .each(({ surveyId, email, choice }) => {
-        Survey.updateOne(
-          {
-            _id: surveyId,
-            recipients: {
-              $elemMatch: { email: email, responded: false }
-            }
-          },
-          {
-            $inc: { [choice]: 1 },
-            $set: { 'recipients.$.responded': true },
-            lastResponded: new Date()
-          }
-        ).exec();
-      })
-      .value();
-
-    res.send({});
+    const result = surveyService.updateSurveys(req.body);
+    res.send(result);
   });
```

Comment:
Since this was a bunch of logic and wrote to the database, it needed moved into a service, so I did that.

It's all replaced by me calling my service method and sending the result along.